src/client/preview.js

```diff
@@ -96,6 +96,11 @@ export function open({hash, eval: compile} = {}) {
         enableCopyButtons();
         break;
       }
+      case "update-image": {
+        const image = document.querySelector(`img[src^="${message.path}"]`);
```

Comment:
This won’t work if the same image is referenced in multiple places. And also this could theory interfere with dynamically generated content by the user. But I don’t think an update-image event should be necessary in any case, since image tags are server-side rendered, and so should already be handled by diffing the generated HTML. That was my point about generalizing the updates of static assets — to some degree it should happen automatically. Stylesheets are special because they get rendered into the head and so aren’t covered by normal diffing.

Reply:
i've swapped in `querySelectorAll`, but it sounds like a different general approach is in order.